frontend/src/components/ViewTrips/create-new-trip.js

```diff
@@ -1,16 +1,125 @@
+ * @param {string} dateStr String containing a date in the form 'yyyy-mm-dd'.
+ * @return {firebase.firestore.Timestamp} Firestore timestamp object created.
  */
-function createTrip(tripObj) {
-  console.log(`trip title: ${tripObj.name}`);
-  console.log(`trip description: ${tripObj.description}`);
-  console.log(`trip destination: ${tripObj.destination}`);
-  console.log(`trip start date: ${tripObj.startDate}`);
-  console.log(`trip end date: ${tripObj.endDate}`);
-  console.log(`trip collaborators: ${tripObj.collaborators}`);
+export function getTimestampFromDateString(dateStr) {
+  const dateParts = dateStr.split('-').map(str => +str);
+  if (dateParts.length === 1 && dateParts[0] === 0) {
+    return firebase.firestore.Timestamp.now();
+  }
+
+  const date = new Date(dateParts[0], dateParts[1] - 1, dateParts[2]);
+  return firebase.firestore.Timestamp.fromDate(date);
+}
+
+/**
+ * Return an array of collaborator uids given the emails provided in the
+ * add trip form.
+ *
+ * @param {!Array{string}} collaboratorEmailsArr Array of emails corresponding
+ *     to the  collaborators of the trip (not including the trip creator email).
+ * @return {!Array{string}} Array of all collaborator uids (including trip
+ *     creator uid).
+ */
+export function getCollaboratorUidArray(collaboratorEmailArr) {
+  collaboratorEmailArr = [getCurUserEmail()].concat(collaboratorEmailArr);
+  while (collaboratorEmailArr.includes('')) {
+    const emptyStrIdx = collaboratorEmailArr.indexOf('');
+    collaboratorEmailArr.splice(emptyStrIdx, 1);
+  }
+  return collaboratorEmailArr.map(userEmail => getUidFromUserEmail(userEmail));
+}
+
+/**
+ * Returns a formatted and cleaned trip object that will be used as the data
+ * for the created Trip document.
+ *
+ * We know that rawTripObj will contain all of the necessary fields because each
+ * key-value pair is explicitly included. This means, only the value
+ * corresponding to each key needs to be checked.
+ * For text element inputs, React has built in protections against injection/XSS
+ * attacks. Thus, no sanitization is needed for text inputs besides providing a
+ * default value in a Trip field where applicable.
+ *
+ * @param {Object} rawTripObj A JS Object containing the raw form data from the
+ *     add trip form.
+ * @return {Object} Formatted/cleaned version of `rawTripObj` holding the data
+ *     for the new Trip document that is to be created.
+ */
+function formatTripData(rawTripObj) {
+  const formattedTripObj =
+  {
+    trip_creation_time: firebase.firestore.Timestamp.now(),
+    name:               getTripName(rawTripObj.name),
+    description:        rawTripObj.description,
+    destination:        getTripDestination(rawTripObj.destination),
+    start_date:         getTimestampFromDateString(rawTripObj.startDate),
+    end_date:           getTimestampFromDateString(rawTripObj.endDate),
+    collaborators:      getCollaboratorUidArray(rawTripObj.collaboratorEmails)
```

Comment:
It will always be a string of some sort. So it will be either an empty string (if they type nothing in) or whatever the user put in the form. This makes the description "optional" and causes no harm if they don't fill it out.